### ISSUE TITLE
Authentication failed despite quadruple checking credentials

### ISSUE THREAD

**Issue by calebcaudill** [ProtonVPN/protonvpn-cli-ng#129]
I have tried several times and am 100% sure I am providing the correct credentials. I am copying and pasting from the openVPN section of my protonmail account. I am using the command to connect to the fastest server available, yet I always get an authentication error. Not sure what to do at this point.

**Comment by Rafficer** [comment]
What ProtonVPN Tier are you on?

Can you post the contents of `~/.pvpn-cli/pvpn-cli.cfg`?

**Comment by fazlerabbi37** [comment]
can confirm this with ProtonVPN-CLI v2.2.2.

---

I tried to investigate a bit the reason behind it and to my astonishment it connected when the OpenVPN username and OpenVPN password were same. Even if a single character case is changed it fails to authenticate. I tried a few more time and it only authenticates when the user name and password are same! The username only supports alpha and numeric so the password is very week!

---

**Log:**
openvpn_connect:462 — connect.ovpn written
disconnect:308 — Initiating disconnect
is_connected:197 — Checking connection Status. OpenVPN processes: 0
manage_dns:611 — Restoring DNS
manage_dns:628 — No Backupfile found
manage_ipv6:687 — Restoring ip6tables
manage_ipv6:702 — No Backupfile found
manage_killswitch:777 — Restoring iptables
manage_killswitch:786 — No Backupfile found
disconnect:348 — No connection found
get_ip_info:143 — Getting IP Information
call_api:36 — Initiating API Call: https://api.protonvpn.ch/vpn/location
call_api:65 — Successful json response
openvpn_connect:482 — OpenVPN process started
openvpn_connect:525 — Authentication failure

**Comment by Rafficer** [comment]
@fazlerabbi37 

I'm not using the same string for username and password and I'm also not using only alphanumeric passwords, so this issue seems to be somewhere else.

Can you check `sudo cat ~/.pvpn-cli/pvpnpass` and see if your username and password in that file are correct?

**Comment by fazlerabbi37** [comment]
> I'm not using the same string for username and password and I'm also not using only alphanumeric passwords, so this issue seems to be somewhere else.

This needs to be verified by more users. I just reported what I saw.

> Can you check sudo cat ~/.pvpn-cli/pvpnpass and see if your username and password in that file are correct?

Yes, they are correct.

**Comment by Rafficer** [comment]
I think it's fairly sure to say that if most users would only be able to use the program if their username and password are the same string, there would already be an outcry.

**Comment by fazlerabbi37** [comment]
> I think it's fairly sure to say that if most users would only be able to use the program if their username and password are the same string, there would already be an outcry.

True. I just can't find any explanation for my cases! :unamused:. If we could find the problem for auth failure we would be able to explain that as well.

**Comment by Rafficer** [comment]
You're using xterm, right? And you've had another issues with xterm as well. Can you maybe try a different terminal emulator, like Gnome Terminal or Konsole?

**Comment by fazlerabbi37** [comment]
Let me try that.

**Comment by Rafficer** [comment]
Please don't just try to connect there, but purge your ProtonVPN configuration via `sudo protonvpn configure` and then reinitialize it with the other terminal.

**Comment by Rafficer** [comment]
@fazlerabbi37 Any news?

**Comment by fazlerabbi37** [comment]
> @fazlerabbi37 Any news?

still same error for me. tried with tilix, terminator, konsole and hyper. :disappointed:

**Comment by Rafficer** [comment]
That's weird. You may want to check with the [support Team](https://protonvpn.com/support-form). I don't know what else this could be.

**Comment by KBeDevel** [comment]
Getting the same result with CLI v2.2.2 in Fedora 31 KDE spin. (I have the Plus plan)

**Comment by Rafficer** [comment]
I'm developing on Fedora KDE, so I doubt that there are any issues with that setup in particular.

Really make sure you're using the OpenVPN credentials, not the ProtonVPN login credentials.

**Comment by KBeDevel** [comment]
Yup, was that @Rafficer

**Comment by calfax** [comment]
@Rafficer  It *does* specifically state this in the configuration but...force of habit and human nature.  For anyone else struggling to connect, you have to enter the OpenVPN credentials not the ProtonVPN credentials (these are at the bottom of the web page when you log into your ProtonVPN account through the browser).

**Comment by AlekseiMarinichenko** [comment]
I am getting randomly the message.

> [!] Authentication failed. 
> [!] Please make sure that your Username and Password is correct.

Sometimes it connects, sometimes it does not. I am just waiting and running a "connect" command again and it connects. 

Any ideas?

**Comment by Rafficer** [comment]
@AlekseiMarinichenko what Plan do you have configured? Make sure it properly matches your actual ProtonVPN plan (they are not yet automatically synchronized).

If you are a basic user and try to connect to a plus server, that's exactly what happens.

**Comment by AlekseiMarinichenko** [comment]
Free plan.

> If you are a basic user and try to connect to a plus server, that's exactly what happens.

Not in my case. I use `protonvpn c -f` or `protonvpn c -r`, so I am not connecting to the "plus" servers.

The whole my usage of VPN is like:
- VPN connection drops (it is really often)
- I am trying to reconnect 
- getting "There was an error connecting to the ProtonVPN API." or "Authentication failed."
- waiting 
- reconnecting manually again...


While "There was an error connecting to the ProtonVPN API." could be because of free servers high load, but "Authentication failed." is really annoying and stops me from upgrading to a paid plan.

**Comment by Rafficer** [comment]
If you have the wrong plan configured, -f and -r won't sort these servers out.

Please make sure you got free configured.

**Comment by AlekseiMarinichenko** [comment]
> If you have the wrong plan configured, -f and -r won't sort these servers out.

If so, why do I only see the messages like "NL-FREE#3" or "JP-FREE#2" (with a FREE word)?

**Comment by AlekseiMarinichenko** [comment]
> If you have the wrong plan configured, -f and -r won't sort these servers out.

If so, why do I only see the messages like "Connecting to NL-FREE#2 via UDP..." (only with **FREE** word included)?

> Please make sure you got free configured.

Ok, I initialized everything from scratch. I will test if it helps.

**Comment by AlekseiMarinichenko** [comment]
Configuration from scratch did not help. I am again disconnecting randomly and randomly getting a messages like when trying to connect/reconnect:

```
There is already a VPN connection running.
Terminating previous connection...
Connecting to NL-FREE#3 via UDP...
[!] Authentication failed. 
[!] Please make sure that your Username and Password is correct.
```
In short, everything is still as I described in the comment above https://github.com/ProtonVPN/linux-cli/issues/129#issuecomment-621943833

**Comment by Rafficer** [comment]
The only other issue could be that the server still thinks for a short time that a connection is active, and since you're only allowed one simultaneous connection that's what happens.

A paid account would very likely solve this.

**Comment by AlekseiMarinichenko** [comment]
> The only other issue could be that the server still thinks for a short time that a connection is active, and since you're only allowed one simultaneous connection that's what happens.
> 
> A paid account would very likely solve this.

Probably. Thanks, I will try a paid account then.

**Comment by hexpwn** [comment]
I am having the same issue. I have 2Fa enabled and I think this may be the root cause... Is there any way to connect using 2FA?
My plan is Plus.
